analyzer/src/main/kotlin/managers/GoMod.kt

```diff
@@ -169,7 +169,12 @@ class GoMod(
             graph.addEdge(parent, child)
         }
 
-        graph = graph.subgraph(getUsedPackages(graph, projectDir, graph.projectId()))
+        val usedPackages = getUsedPackages(graph, projectDir, graph.projectId())
+        if (usedPackages.size < graph.size()) {
+            log.debug { "Removing ${graph.size() - usedPackages.size} unused packages from the dependency graph." }
```

Comment:
"Removed" would be better because it already happened.

Reply:
Out of scope, so done it in a separate commit on top.